File: leetcode-c/lc_026.c
//
// LeetCode
// 26. Remove Duplicates from Sorted Array
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
//


int removeDuplicates(int* nums, int numsSize) {
    int i = 0, j;

    if (numsSize == 0)
        return 0;

    for (j = 1; j < numsSize; j++)
        if (nums[j] != nums[i]) {
            i++;
            nums[i] = nums[j];
        }

    return i + 1;
}

File: leetcode-go/lc_026.go
//
// LeetCode
// 26. Remove Duplicates from Sorted Array
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
//


func removeDuplicates(nums []int) int {
    if len(nums) <= 1 {
        return len(nums)
    }

    idx := 0
    for i:= 0; i < len(nums); i++ {
        if nums[i] != nums[idx] {
            idx++
            nums[idx] = nums[i]
        }
    }

    return idx + 1
}
